compiler/ballerina-parser/src/main/java/io/ballerina/compiler/internal/parser/BallerinaParser.java

```diff
@@ -301,6 +301,29 @@ public STNode parseAsObjectMember() {
         return objectMember;
     }
 
+    public STNode parseAsIntermediateClause(boolean isRhsExpr, boolean allowActions) {
+        STNode intermediateClause = null;
+        startContext(ParserRuleContext.QUERY_EXPRESSION);
+        startContext(ParserRuleContext.VAR_DECL_STMT);
+        startContext(ParserRuleContext.FUNC_BODY_BLOCK);
+        startContext(ParserRuleContext.FUNC_DEF);
+        startContext(ParserRuleContext.COMP_UNIT);
+        boolean isEndOfIntermediateClause = isEndOfIntermediateClause(peek().kind);
+        if (!isEndOfIntermediateClause) {
+            intermediateClause = parseIntermediateClause(isRhsExpr, allowActions);
+        }
+        if (intermediateClause == null) {
+            intermediateClause = SyntaxErrors.createMissingToken(SyntaxKind.SELECT_CLAUSE);
```

Comment:
This is incorrect we need to create a missing intermediate clause. Let's create a missing where clause node